File: result/translate/starchat/java2cpp/c++/MAXIMUM_CONSECUTIVE_NUMBERS_PRESENT_ARRAY.cpp
int f_gold ( int arr [ ], int n ) {
    set < int > S;
    for ( int i = 0 ; i < n ; i ++ ) S. insert ( arr [ i ] ) ;
    int ans = 0 ;
    for ( int i = 0 ; i < n ; i ++ ) {
        if ( S. find ( arr [ i ] )!= S. end ( ) ) {
            int j = *S. lower_bound ( arr [ i ] ) ;
            while ( S. find ( j )!= S. end ( ) ) j ++ ;
            ans = max ( ans, j - arr [ i ] ) ;
        }
    }
    return ans ;
}

File: result/translate/chatgpt/java2python/python/MAXIMUM_CONSECUTIVE_NUMBERS_PRESENT_ARRAY.py
def f_gold(arr, n):
    S = set()
    for i in range(n):
        S.add(arr[i])
    ans = 0
    for i in range(n):
        if arr[i] in S:
            j = arr[i]
            while j in S:
                j += 1
            ans = max(ans, j - arr[i])
    return ans